test/job-history/get_jobhistory_hdfs.py

```diff
@@ -0,0 +1,795 @@
+            return data
+        if data in SPARK_SAFE_VALUES:
+            return data
+        try:
+            int(data)
+            # Must be an integer
+            return data
+        except ValueError:
+            # Must not have been an integer
+            pass
+        return _anonymize_str(data)
+    else:
+        raise NotImplementedError("Unknown anonymization type {}".format(data))
+
+
+def _anonymize_mapreduce_file(path):
+    """Anonymize a mapreduce history file.
+
+    @param str path: The local path to the file.
+    """
+    with open(path) as f:
+        input_lines = f.readlines()
+    with open(path, "w") as f:
+        print >>f, input_lines[0].rstrip()
+        for input_line in input_lines[1:]:
+            if input_line != "\n":
+                data = json.loads(input_line)
+                data = _anonymize_mapreduce_data(data)
+                print >>f, json.dumps(data)
+            else:
+                print >>f, ""
+
+
+def _anonymize_mapreduce_data(data, parent_info=None):
+    """Anonymize a decoded json mapreduce job history object.
+
+    @param object data: An object that may contain structure in dict or lists or be a scalar.
+    @param parent_info data: A parent information object, sometimes useful for anonymization
+
+    @rtype: object
+
+    Returns an anonymized version of the same structure.
+    """
+    if isinstance(data, dict):
+        return {k: _anonymize_mapreduce_data(v, k) for k, v in data.iteritems()}
+    elif isinstance(data, list):
+        return [_anonymize_mapreduce_data(v) for v in data]
+    elif isinstance(data, NUMBER_TYPES):
+        return data
+    elif isinstance(data, STRING_TYPES):
+        if parent_info in MAPREDUCE_SAFE_PARENTS:
+            return data
+        if data in MAPREDUCE_SAFE_VALUES:
+            return data
+        if _is_str_number(data):
+            return data
+        return _anonymize_str(data)
+    elif data is None:
+        return None
+    else:
+        raise NotImplementedError("Unknown anonymization type [{}]".format(data))
+
+
+def _anonymize_xml_conf_file(path):
+    """Anonymize an configuration (xml) file.
+
+    @param str path: The local path to the file.
+    """
+    tree = etree.parse(path)
+    root = tree.getroot()
+    for sub_element in root:
+        if sub_element.find("name") is None:
+            continue
+        name = sub_element.find("name").text
+        value = sub_element.find("value").text
+        if _is_str_number(value):
+            continue
+        if name in XML_CONF_SAFE_KEYS:
+            continue
+        if value in XML_CONF_SAFE_VALUES:
+            continue
+        new_value = _anonymize_str(value)
+        sub_element.find("value").text = new_value
+    tree.write(path)
+
+
+def _anonymize_str(plaintext):
+    """Anonymize a given string.
+
+    @param str plaintext: The plaintext string to anonymize.
+    @rtype: str
+    """
+    hasher = hashlib.sha256()
+    hasher.update(plaintext)
```

Comment:
A pedantic comment on security: it might be a good idea to add a random salt to the plaintext before hashing. Otherwise low entropy plaintexts are crackable. I see two ways of doing this:
1. Take the salt as an input to the script. This would allow an organization to use a single salt for all their files maintaining plaintext correlations between runs.
2. Generate the salt at the beginning of each script run and use it for the duration of the run. This is easier but correlations between runs are lost.
Also, of course there could always be the option to do either.